Mappers/HttpMapper.php

```diff
@@ -1,226 +1,237 @@
+    public function delete(string $byPropertyName, $propertyValue): bool
     {
-        $content = $this->getHttpClient()->sendHttpRequest($action, $url, $options)->getBody();
-        
-        if($content === null){
-            return true;
-        }
+        $this->sendRequest($this->requestFactory->create(
+            self::DELETE, $this->urlRender->prepareDeleteUrl([$byPropertyName => $propertyValue])
+        ));
 
-        if(array_key_exists('items', $content)){
-            return $this->buildPaginationCollection($content);
-        }
-
-        return $this->buildObject($content);
+        return true;
     }
 
-    /**
-     * @deprecated
-     * @param $action
-     * @param $url
-     * @param $options
-     * @return ArraySerializable|\Modules\LibraryModule\Models\Collection|void
-     * @throws EnergonException
-     */
-    protected function sendHttpRequestForGetStatus($action, $url, $options)
-    {
-        return $this->getHttpClient()->sendHttpRequestForGetStatus($action, $url, $options);
-    }
 
     /**
-     * @param ArraySerializable $object
-     * @param $identifierName
-     * @param $urlName
-     * @return string
+     * @param array $objectData
+     * @return Collection|mixed
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    protected function getUrlWithObjectAndIdentifier(ArraySerializable $object, $identifierName, $urlName)
+    public function updatePatch(array $objectData)
     {
-        $identifier = $object->toArray()[$identifierName];
-        return vsprintf($this->getHttpMapperConfig($urlName), [$identifier]);
+        return $this->getModel(
+            $this->requestFactory->create(self::PATH, $this->urlRender->prepareUpdateUrl($objectData), [], $objectData)
+        );
     }
 
     /**
-     * @param $identifierValue
-     * @param $url
-     * @return string
+     * @param RequestInterface $request
+     * @return \Infrastructure\Models\Http\ResponseInterface
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    private function getUrlWithIdentifier($identifierValue, $url)
+    protected function sendRequest(RequestInterface $request)
     {
-        $urlTemplate = $this->getHttpMapperConfig($url);
-        return vsprintf($urlTemplate, [$identifierValue]);
+        return $this->getHttpClient()->send($this->mergeDefaultData($request));
     }
 
     /**
-     * @return HttpClient
+     * @param RequestInterface $request
+     * @return mixed
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    protected function getHttpClient()
+    protected function getModel(RequestInterface $request): ArraySerializable
```

Comment:
please  Change  methods naming. SendRequestForEntity to be more clear or smth like this.

Reply:
change name